fix: work js-api signature

mahongran opened the pull request.

mahongran committed:
feat: enhance WorkAccessToken to include AgentID for improved token management

- Added AgentID field to WorkAccessToken struct.
- Updated NewWorkAccessToken function to accept AgentID as a parameter.
- Modified access token cache key to incorporate AgentID, ensuring unique cache entries per agent.

This change improves the handling of access tokens in a multi-agent environment.

mahongran committed:
refactor: enhance WorkAccessToken to improve cache key handling

- Updated the AgentID field in WorkAccessToken struct to clarify its optional nature for distinguishing applications.
- Modified the access token cache key construction to support both new and legacy formats based on the presence of AgentID.
- Added comments for better understanding of the cache key logic and its compatibility with historical versions.

This change improves the flexibility and clarity of access token management in multi-agent scenarios.

mahongran committed:
feat(work): add JsSdk method for JavaScript SDK integration

- Introduced a new JsSdk method in the Work struct to facilitate the creation of a Js instance.
- This addition enhances the functionality of the Work module by enabling JavaScript SDK support.

This change improves the integration capabilities for developers working with the WeChat Work API.

mahongran committed:
Merge branch 'v2' into work-js-sdk

mahongran committed:
Merge remote-tracking branch 'upstream/v2' into work-js-sdk

# Conflicts:
#	credential/default_access_token.go

mahongran committed:
Merge remote-tracking branch 'origin/work-js-sdk' into work-js-sdk

# Conflicts:
#	credential/default_access_token.go

mahongran committed:
fix gofmt

mahongran committed:
refactor(jsapi): simplify signature generation in GetConfig and GetAgentConfig methods

- Replaced the use of strconv to format the signature string with fmt.Sprintf for improved readability.
- Updated the signature generation logic in both GetConfig and GetAgentConfig methods to use a single formatted string.

This change enhances code clarity and maintains functionality in the signature generation process.

mahongran committed:
Merge remote-tracking branch 'upstream/v2' into v2

# Conflicts:
#	credential/default_access_token.go

mahongran committed:
Merge branch 'work-js-sdk' into v2

# Conflicts:
#	credential/default_access_token.go

mahongran committed:
Merge branch 'v2' into work-js-sdk

mahongran committed:
fix gofmt

houseme commented on work/jsapi/jsapi.go:
```diff
@@ -1,8 +1,7 @@
 import (
-	"strconv"
-
+	"fmt"
```
4,5之间有空行

mahongran committed:
fix gofmt

mahongran committed:
fix gofmt

mahongran committed:
fix gofmt

mahongran commented:
fmt 下面要加上空行才可以通过 check 好奇怪

houseme commented:
> fmt 下面要加上空行才可以通过 check 好奇怪

是官方包、第三方包、项目包路径都是按组引入，中间使用空行分隔。

houseme commented on officialaccount/js/js.go:
```diff
@@ -50,12 +50,10 @@ func (js *Js) GetConfig(uri string) (config *Config, err error) {
 	timestamp := util.GetCurrTS()
 	str := fmt.Sprintf("jsapi_ticket=%s&noncestr=%s&timestamp=%d&url=%s", ticketStr, nonceStr, timestamp, uri)
 	sigStr := util.Signature(str)
-
```
42行是不是顺带优化一下，移到56行之后再初始化呢？

mahongran replied:
ok

mahongran committed:
refactor(js): correct initialization of Config in GetConfig method

- Removed redundant initialization of the Config variable and ensured it is instantiated correctly before use.
- This change improves code clarity and maintains the intended functionality of the GetConfig method.
